src/containers/target-highlight.jsx

```diff
@@ -0,0 +1,78 @@
+import bindAll from 'lodash.bindall';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {connect} from 'react-redux';
+import VM from 'scratch-vm';
+
+class TargetHighlight extends React.Component {
+    constructor (props) {
+        super(props);
+        bindAll(this, [
+            'getPageCoords'
+        ]);
+    }
+
+    // Transform scratch coordinates into page coordinates
+    getPageCoords (x, y) {
+        const nativeSize = this.props.vm.renderer.getNativeSize();
+        const rect = this.props.vm.renderer.canvas.getBoundingClientRect();
```

Comment:
Since this forces a reflow, could it cause performance issues if it's being called in render? Also, we already do this and cache it in the stage container. To avoid doing this more than we need to, maybe this component should receive it as a prop? https://github.com/LLK/scratch-gui/blob/develop/src/containers/stage.jsx#L139